components/dashboard/src/projects/NewProject.tsx

```diff
@@ -0,0 +1,314 @@
+            reconfigure();
+        } else {
+            setSelectedAccount(value);
+        }
+    }
+
+    const toSimpleName = (fullName: string) => {
+        const splitted = fullName.split("/");
+        if (splitted.length < 2) {
+            return fullName;
+        }
+        return splitted.shift() && splitted.join("/");
+    }
+
+    const selectProvider = async (ap: AuthProviderInfo) => {
+        const token = await getToken(ap.host);
+        if (token) {
+            setShowGitProviders(false);
+            setProvider(ap.host);
+            return;
+        }
+        await openAuthorizeWindow({
+            host: ap.host,
+            scopes: ap.requirements?.default,
+            onSuccess: () => {
+                setShowGitProviders(false);
+                setProvider(ap.host);
+            },
+            onError: (error) => {
+                console.log(error);
+            }
+        });
+    }
+
+    const reposToRender = Array.from(reposInAccounts).filter(r => r.account === selectedAccount);
+    const accounts = Array.from(new Set(Array.from(reposInAccounts).map(r => r.account)));
+
+    const renderSelectRepository = () => (<>
+        <h3 className="pb-2 mt-8">Select Repository</h3>
+
+        <div className="mt-8 w-96 border rounded-xl border-gray-100 flex-col">
+            {showGitProviders && (<>
+                <div className="p-6">
+                    <div className="text-center text-gray-500">
+                        Select a Git provider first and continue with your repositories.
+                    </div>
+                    <div className="mt-6 flex flex-col space-y-3 items-center">
+                        {authProviders.map(ap => {
+                            return (
+                                <button key={"button" + ap.host} className="btn-login flex-none w-56 h-10 p-0 inline-flex" onClick={() => selectProvider(ap)}>
+                                    {iconForAuthProvider(ap.authProviderType)}
+                                    <span className="pt-2 pb-2 mr-3 text-sm my-auto font-medium truncate overflow-ellipsis">Continue with {simplifyProviderName(ap.host)}</span>
+                                </button>
+                            );
+                        })}
+                    </div>
+                </div>
+            </>)}
+            {!showGitProviders && (<>
+                <div className="p-6">
+                    <div className="flex flex-col space-y-2">
+                        <label htmlFor="selectAccount" className="font-medium">Account</label>
+                        <select name="selectAccount" value={selectedAccount} className="w-full"
+                            onChange={(e) => onSelectAccount(e.target.value)}>
+                            {accounts.map(a => (<option key={`account-${a}`} value={a}>{`${a} (${provider})`}</option>))}
+                            {isGitHub() && (<option value="addOrg">Add GH Org</option>)}
```

Comment:
**nitpick**: Reminder to avoid acronyms like GH here. 😇

Reply:
Very well. Just realized how surprising the comments are, while changing code at the very same time ;-)